routes/cube_routes.js

```diff
@@ -682,56 +673,8 @@ router.get('/compare/:id_a/to/:id_b', async (req, res) => {
     const cubeBq = Cube.findOne(build_id_query(id_b));
 
     const [cubeA, cubeB] = await Promise.all([cubeAq, cubeBq]);
-
-    const pids = new Set();
-    [cubeA, cubeB].forEach((cube) => {
-      cube.cards.forEach(function(card, index) {
-        card.details = {
-          ...carddb.cardFromId(card.cardID),
-        };
-        if (!card.type_line) {
-          card.type_line = card.details.type;
-        }
-        if (card.details.tcgplayer_id) {
-          pids.add(card.details.tcgplayer_id);
-        }
-      });
-    });
-
-    const price_dict = await GetPrices([...pids]);
-    [cubeA, cubeB].forEach((cube) => {
-      cube.cards.forEach(function(card, index) {
-        if (card.details.tcgplayer_id) {
-          if (price_dict[card.details.tcgplayer_id]) {
-            card.details.price = price_dict[card.details.tcgplayer_id];
-          }
-          if (price_dict[card.details.tcgplayer_id + '_foil']) {
-            card.details.price_foil = price_dict[card.details.tcgplayer_id + '_foil'];
-          }
-        }
-      });
-    });
-
-    let in_both = [];
-    let only_a = cubeA.cards.slice(0);
-    let only_b = cubeB.cards.slice(0);
-    let a_names = only_a.map((card) => card.details.name);
-    let b_names = only_b.map((card) => card.details.name);
-
-    cubeA.cards.forEach(function(card, index) {
-      if (b_names.includes(card.details.name)) {
-        in_both.push(card);
-
-        only_a.splice(a_names.indexOf(card.details.name), 1);
-        only_b.splice(b_names.indexOf(card.details.name), 1);
-
-        a_names.splice(a_names.indexOf(card.details.name), 1);
-        b_names.splice(b_names.indexOf(card.details.name), 1);
-      }
-    });
-
-    let all_cards = in_both.concat(only_a).concat(only_b);
-
+    await populateCardDetails([cubeA.cards, cubeB.cards], carddb, prices);
```

Comment:
Modifying collections passed in as a side effect is an anti-pattern, please return the values you need and don't modify the passed in data